Adaptors/S3/src/ObjectStorage.cs

```diff
@@ -0,0 +1,249 @@
+    }
+
+    // Get the data from the response stream
+    using var reader      = new StreamReader(response.ResponseStream);
+    var       fileContent = reader.ReadToEnd();
+
+    var valuesCount = int.Parse(fileContent!);
+
+    if (valuesCount == 0)
+    {
+      yield break;
+    }
+
+    foreach (var chunkTask in Enumerable.Range(0,
+                                               valuesCount)
+                                        .Select(index => s3Client_.StringByteGetAsync(bucketName_,
+                                                                                      $"{objectStorageName_}{key}_{index}",
+                                                                                      logger_))
+                                        .ToList())
+    {
+      yield return (await chunkTask.ConfigureAwait(false))!;
+    }
+  }
+
+  /// <inheritdoc />
+  public async Task<bool> TryDeleteAsync(string            key,
+                                         CancellationToken cancellationToken = default)
+  {
+    using var _ = logger_.LogFunction(objectStorageName_ + key);
+    var value = await s3Client_.StringGetValueAsync(bucketName_,
+                                                    $"{objectStorageName_}{key}_count",
+                                                    logger_)
+                               .ConfigureAwait(false);
+
+    if (value == null)
+    {
+      throw new ObjectDataNotFoundException("Key not found");
+    }
+
+    var valuesCount = int.Parse(value!);
+    var keyList = Enumerable.Range(0,
+                                   valuesCount)
+                            .Select(index => new KeyVersion
+                                             {
+                                               Key = $"{objectStorageName_}{key}_{index}",
+                                             })
+                            .Concat(new[]
+                                    {
+                                      new KeyVersion
+                                      {
+                                        Key = $"{objectStorageName_}{key}_count",
+                                      },
+                                    })
+                            .ToList();
+    var multiObjectDeleteRequest = new DeleteObjectsRequest
+                                   {
+                                     BucketName = bucketName_,
+                                     Objects    = keyList,
+                                   };
+
+    var deleteObjectsResponse = await s3Client_.DeleteObjectsAsync(multiObjectDeleteRequest)
+                                               .ConfigureAwait(false);
+    return deleteObjectsResponse.DeletedObjects.Count == valuesCount + 1;
+  }
+
+  /// <inheritdoc />
+  public IAsyncEnumerable<string> ListKeysAsync(CancellationToken cancellationToken = default)
+    => throw new NotImplementedException();
+}
+
+internal static class S3StorageHelper
```

Comment:
This is an extension class for `AmazonS3Client ` so it should be called `AmazonS3ClientExt`